### DISCUSSION TITLE
To detect unused function parameter.

### DISCUSSION THREAD

**Discussion by anzosasuke** [NationalSecurityAgency/ghidra#4138] [Q&A] [answered]
I was looking to get the unused parameter of function. I am thinking getting getForwardslice() to get the information( I did something but didnt make any sense ). I also have getDescendents() from High variable of Pcode of a function. Will you guys , may be share some thoughts on how I could achieve my goal?
And How do I read a Pcode, is there any documentation or supplements that I could use?

**Comment by ghidracadabra** [comment, marked as answer]
For a HighFunction `h`, `h.getLocalSymbolMap().getParam(i)` with get the ith parameter.  You can then use the `getInstances()` method to return all varnodes which that parameter corresponds to.  For each varnode, you can use the `getDescendants()` method to get all of the pcode ops taking it as input.

You can find pcode documentation in the file `<ghidra install dir>/docs/languages/html/pcoderef.html`
